announcement: Add move to Docker 25 banner (#343)
flatcar/flatcar-website: content/releases/_index.md

title: "Releases"
date: 2018-04-17T17:32:59+02:00
draft: false
announcement: true
announcement_bg: "#12172b"
announcement_text_class: "text-light"
announcement_title: "Move to Docker 25"
announcement_message: "

> We will begin moving to **Docker 25** in the near future. Docker 25 and above **remove the devicemapper** storage driver. While new provisionings should not be affected by the change, long-lived nodes which use these back-ends will break after the update.

Reviewer: ```suggestion
We will begin moving to **Docker 25 (or newer)** in the near future. Among other changes, Docker 25 and above **remove the devicemapper** storage driver. While new provisionings should not be affected by the change, long-lived nodes which use this back-end will break after the update.
```

> *  We plan to introduce Docker 25 in the **Alpha release late July 2024**.

Reviewer: ```suggestion
We plan to introduce Docker 25 (or a newer release) in the **Alpha release late July 2024**.
```

> We plan to introduce Docker 25 (or a newer release) in the **Alpha release late July 2024**.

Reviewer: ```suggestion
* We plan to introduce Docker 25 (or a newer release) in the **Alpha release late July 2024**.
```

> *  We expect Docker 25 to **hit Stable in October 2024 the earliest**.

Reviewer: ```suggestion
*  We expect the Docker upgrade to **hit Stable in October 2024 the earliest**.
```

Any nodes using the devicemapper storage driver will lose access to all docker state (local container images and stopped containers) after this update. Please *participate in Beta testing and run Beta canaries* if you suspect you might be affected.
If you are reading this after Docker 25 hits stable in late 2024 and want to keep using Docker 24 while still updating to the latest OS release, please consider masking Docker 25 altogether and using the Docker 24 sysext from our [sysext-bakery](https://github.com/flatcar/sysext-bakery?tab=readme-ov-file#systemd-sysext).

> "

Reviewer: ```suggestion
Please find a full, up-to-date list of deprecated and removed features across Docker versions here: [https://github.com/docker/cli/blob/master/docs/deprecated.md](https://github.com/docker/cli/blob/master/docs/deprecated.md)

We will also discuss the docker upgrade and provide status updates in our [Office Hours](https://github.com/flatcar/Flatcar/discussions/categories/flatcar-office-hours?discussions_q=category%3A%22Flatcar+Office+Hours%22+is%3Aopen) and [Developer Sync](https://github.com/flatcar/Flatcar/discussions/categories/flatcar-developer-sync%22+is%3Aopen) calls.
```

* We plan to introduce Docker 25 (or a newer release) in the **Alpha release late July 2024**.

*  We expect the Docker upgrade to **hit Stable in October 2024 the earliest**.

Any nodes using the devicemapper storage driver will lose access to all docker state (local container images and stopped containers) after this update. Please *participate in Beta testing and run Beta canaries* if you suspect you might be affected.
If you are reading this after Docker 25 hits stable in late 2024 and want to keep using Docker 24 while still updating to the latest OS release, please consider masking Docker 25 altogether and using the Docker 24 sysext from our [sysext-bakery](https://github.com/flatcar/sysext-bakery?tab=readme-ov-file#systemd-sysext).
Please find a full, up-to-date list of deprecated and removed features across Docker versions here: [https://github.com/docker/cli/blob/master/docs/deprecated.md](https://github.com/docker/cli/blob/master/docs/deprecated.md)

> We will also discuss the docker upgrade and provide status updates in our [Office Hours](https://github.com/flatcar/Flatcar/discussions/categories/flatcar-office-hours?discussions_q=category%3A%22Flatcar+Office+Hours%22+is%3Aopen) and [Developer Sync](https://github.com/flatcar/Flatcar/discussions/categories/flatcar-developer-sync%22+is%3Aopen) calls.

Reviewer: Good catch, thank you!
